vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +95,283 @@ def process_shapes(
-    has_eq_side_lengths: bool = compare_side_len_eq(lengths)
-    good_aspect_ratio: bool = filtering.test_min_area_box(approx, SHAPE_ASPECT_RATIO_RANGE)
-
-    angles: npt.NDArray[npt.Shape["*"], npt.Float64]
-    shape: chars.ODLCShape | None = None
-    match len(approx):
-        case 3:
-            shape = chars.ODLCShape.TRIANGLE
-        case 4:
-            angles = get_angles(approx)
-            # if all angles approximately 90 deg, square or rectangle, else trapezoid
-            if np.all(compare_angles(angles, 90, RIGHT_ANG_THRESH)):
-                # if all side lengths are (approximately) equal then square, else rectangle
-                if has_eq_side_lengths and good_aspect_ratio:
-                    shape = chars.ODLCShape.SQUARE
-                else:
-                    shape = chars.ODLCShape.RECTANGLE
-            else:
-                shape = chars.ODLCShape.TRAPEZOID
-        case 5:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.PENTAGON
-        case 6:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.HEXAGON
-        case 7:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.HEPTAGON
-        case 8:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.OCTAGON
-        case 10:
-            if has_eq_side_lengths:
-                shape = chars.ODLCShape.STAR
-        case 12:
-            angles = get_angles(approx)
-            # a plus shape should have exactly 8 right angles *inside* of it (2 on each bar)
-            if np.count_nonzero(compare_angles(angles, 90, RIGHT_ANG_THRESH)) == 8:
-                shape = chars.ODLCShape.CROSS
-        case _:
-            shape = None
-
-    return shape
-
-
-def compare_side_len_eq(lengs: npt.NDArray[npt.Shape["*"], npt.Float64]) -> bool:
+def cartesian_array_to_polar(
+    cartesian_array: NDArray[Shape["*,2,2"], Float64]
+) -> NDArray[Shape["*,2"], Float64]:
     """
-    Checks if all of the side lengths given are (approximately) equal.
+    Converts an array of rectangular (cartesian) coordinates to an array of polar coordinates
 
     Parameters
     ----------
-    lengths : npt.NDArray[npt.Shape["*"], npt.Float64]
-        An array of all the side lengths of a polygon (from get_lengths())
+    arr : List[float]
+        An array of rectangular (cartesian) coordinates
 
     Returns
     -------
-    eq_side_lengths : bool
-        True if all of the given side lengths are approximately equal
+    shape : chars.ODLCShape | None
+        Returns an array of polar coordinates
     """
-    return bool(
-        np.all(np.where(abs((lengs / np.mean(lengs)) - 1) < SIDE_LEN_EQ_THRESH, True, False))
+    cartesian_array_x: NDArray[Shape["*, 2"], Float64] = cartesian_array[:, 0, 1]
+    cartesian_array_y: NDArray[Shape["*, 2"], Float64] = cartesian_array[:, 0, 0]
+    polar_array: NDArray[Shape["2, *"], Float64] = cartesian_to_polar(
```

Comment:
`cartesian_to_polar()` returns a Tuple, not an array. This could probably just be replaced with:
```suggestion
    rho_array = np.hypot(cartesian_array_x, cartesian_array_y)
    phi_array = np.arctan2(cartesian_array_y, cartesian_array_x)
    polar_array = np.stack((rho_array, phi_array))
```
This should mean that this `polar_array` should already be in the needed format, so the swapaxes is no longer necessary. Also add typehints if you use this suggestion

Reply:
I added axis=1 argument to np.stack, because without it it would add all the x-axis items to one array and all the y-axis items to the other.